pkg/hpsf/hpsfTypes.go

```diff
@@ -620,15 +622,27 @@ func (h *HPSF) isSourceComponent(c *Component, connType ConnectionType) bool {
 	return false
 }
 
-// PipelineWithConnectionType is designed to hold a linear set of components
-// connected by a specific connection type. We generate all possible pipelines
-// as a slice of this type.
-type PipelineWithConnectionType struct {
-	ConnType ConnectionType
-	Pipeline []*Component
+// PipelineWithConnections is designed to hold a linear set of components
+// connected by a specific connection type, along with the specific
+// sets of connections used. We generate all possible pipelines
+// as a slice of these.
+type PipelineWithConnections struct {
```

Comment:
What does "pipeline" mean here? Is it the Pipeline Installation, the collector pipeline, or some other thing?

Reply:
See above; I've renamed to "path".